sigopt/xgboost/run.py

```diff
@@ -0,0 +1,109 @@
+import platform
+import copy
+
+import xgboost
+# pylint: disable=no-name-in-module
+from xgboost import DMatrix
+
+from ..context import Context
+from ..log_capture import SystemOutputStreamMonitor
+from .. import create_run
+
+DEFAULT_EVALS_NAME = 'Test Set'
+DEFAULT_RUN_OPTIONS = {
+  'log_sys_info': True,
+  'log_stdout': True,
+  'log_stderr': True,
+  'run': None
+}
+XGB_ALIASES = \
+  {
+     'learning_rate': 'eta',
+     'min_split_loss': 'gamma',
+     'reg_lambda': 'lambda',
+     'reg_alpha': 'alpha'
+  }
+
+
+def parse_run_options(run_options):
+  if run_options:
+    assert run_options.keys() <= DEFAULT_RUN_OPTIONS.keys(), 'Unsupported argument inside run_options'
```

Comment:
I didn't know you could compare dict like this.

Reply:
stackoverflow is too good!